sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_document_retrieval/_document_retrieval.py

```diff
@@ -0,0 +1,467 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import math
+import operator
+from itertools import starmap
+from typing import Dict, List, TypedDict, Tuple, Optional
+from azure.ai.evaluation._evaluators._common import EvaluatorBase
+from azure.ai.evaluation._exceptions import EvaluationException
+from typing_extensions import override, overload
+
+
+RetrievalGroundTruthDocument = TypedDict(
+    "RetrievalGroundTruthDocument", {"document_id": str, "query_relevance_label": int}
+)
+
+RetrievedDocument = TypedDict(
+    "RetrievedDocument", {"document_id": str, "relevance_score": float}
+)
+
+
+class DocumentRetrievalEvaluator(EvaluatorBase):
+    """
+    Calculate document retrieval metrics, such as NDCG, XDCG, Fidelity, Top K Relevance and Holes.
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/evaluation_samples_evaluate.py
+            :start-after: [START document_retrieval_evaluator]
+            :end-before: [END document_retrieval_evaluator]
+            :language: python
+            :dedent: 8
+            :caption: Initialize and call a Document RetrievalEvaluator
+
+    .. admonition:: Example with Threshold:
+        .. literalinclude:: ../samples/evaluation_samples_threshold.py
+            :start-after: [START threshold_document_retrieval_evaluator]
+            :end-before: [END threshold_document_retrieval_evaluator]
+            :language: python
+            :dedent: 8
+            :caption: Initialize with threshold and call a DocumentRetrievalEvaluator.
+    """
+
+    def __init__(
+        self,
+        *,
+        groundtruth_label_min: int = 0,
+        groundtruth_label_max: int = 4,
+        threshold: Optional[dict] = None,
+    ):
+        super().__init__()
+        self.k = 3
+        self.xdcg_discount_factor = 0.6
+
+        if groundtruth_label_min >= groundtruth_label_max:
```

Comment:
Please udpate it to ground_truth_label_min

Reply:
Updated it here, and also in a few other customer-facing places.  I left most of the internal references alone just to avoid too much change at once, but let me know if you think any others should be updated.